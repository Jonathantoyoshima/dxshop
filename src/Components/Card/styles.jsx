import styled from 'styled-components';

export const Root = styled.div`
    align-items: center;
    background-color: ${(props) =>
		props.theme.color.basic.white};
	box-shadow: 5px 5px 15px 5px rgba(0, 0, 0, 0.5);
	width: 100%;
	display: flex;
	padding: 12px 18px;
	margin-bottom: 6px;
`;

export const Title = styled.div`
	align-items: center;
	color: ${(props) => props.theme.color.basic.black};
	display: flex;
	font-size: 1.2rem;
	flex: 1;
`;
