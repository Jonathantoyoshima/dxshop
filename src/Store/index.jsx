import React from 'react';

const categories = [
  { id: 0, name: 'Bebidas' },
  { id: 1, name: 'Doces' },
  { id: 2, name: 'Salgados' },
];

const data = [
  { id: 0, idCategory: 0, name: 'Coca-cola lata 350ml', price: 3.5 },
  { id: 1, idCategory: 0, name: 'Fanta lata 350ml', price: 3.5 },
  { id: 2, idCategory: 1, name: 'Bolinho Ana maria', price: 1.5 },
  { id: 3, idCategory: 2, name: 'Ruffes Pequeno 350g', price: 5.0 },
];

const cart = [];

const info = { data: data, categories: categories, cart: cart, chooseCategory: -1 };

export const StoreContext = React.createContext(info);

export const Store = ({ children }) => {
  const [context, setContext] = React.useState(info);

  return <StoreContext.Provider value={[context, setContext]}>{children}</StoreContext.Provider>;
};
