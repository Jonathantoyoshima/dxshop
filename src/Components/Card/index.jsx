import { Root, Title } from './styles';
import { Button } from 'Styles';

const Card = ({ title, link }) => {
  return (
    <Root>
      <Title>{title}</Title>
      <Button>Adicionar ao carrinho</Button>
    </Root>
  );
};

Card.defaultProps = {
  theme: {
    color: {
      basic: {
        black: '#0c0c0c',
        white: '#fff',
      },
    },
  },
};

Card.defaultProps = {
  theme: {
    color: {
      basic: {
        black: '#0c0c0c',
        white: '#fff',
      },
    },
  },
};

export default Card;
