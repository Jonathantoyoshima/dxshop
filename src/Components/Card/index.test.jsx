import React from 'react';
import { shallow } from 'enzyme';
import { Store } from 'Store';
import Card from './index';

test('Render Build Card Snapshot', () => {
  const component = shallow(
    <Store>
      <Card />
    </Store>
  );
  expect(component).toMatchSnapshot();
});
