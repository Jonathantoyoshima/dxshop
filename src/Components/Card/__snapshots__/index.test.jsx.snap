// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Build Card Snapshot 1`] = `ShallowWrapper {}`;
